src/Hint/NoCapitalisms.hs

```diff
@@ -0,0 +1,119 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE LambdaCase #-}
+{-
+    Detect uses of capitalisms
+
+    Only allow up to two consecutive capital letters in top level
+    identifiers.
+
+    Identifiers containing underscores are exempted from thus rule.
+    Identifiers of FFI bindings are exempted from thus rule.
+
+    Locally bound identifiers and module names are not checked.
+
+<TEST>
+data LHsDecl
+class FOO a where -- @Ignore
+class Foo a where getFOO :: Bool
+data Foo = Bar | BAAZ -- @Ignore
+data Foo = B_ar | BAAZ -- @Ignore
+data Foo = Bar | B_AAZ
+data OTPToken = OTPToken -- @Ignore
+data OTP_Token = Foo 
+sendSMS = _ -- @Ignore
+runTLS = _ -- @Ignore
+runTLSSocket = _ -- @Ignore
+runTLS_Socket
+newtype TLSSettings = TLSSettings -- @Ignore
+tlsSettings
+data CertSettings = CertSettings
+tlsServerHooks
+tlsServerDHEParams = _  -- @Ignore
+type WarpTLSException = () -- @Ignore
+get_SMS
+runCI
+foreign import ccall _FIREMISSLES :: IO ()
+getSMS :: IO () -- @Ignore
+gFOO = _ -- @Ignore
+geFOO = _ -- @Ignore
+getFOO = _ -- @Ignore
+</TEST>
+-}
+
+module Hint.NoCapitalisms(noCapitalismsHint) where
```

Comment:
"capitalism"? How about "NoCAPs"?

Reply:
I do not see such a meaning for capitalism in the dictionary.

Reply:
Yeah, "NoCAPs" seems better since it doesn't have political connotations and shouldn't provoke unnecessary discussions, unlike the current name.